tests/perf-system/analyzer/analyzer.py

```diff
@@ -0,0 +1,290 @@
+        plt.plot(block_latency, throughput_per_block)
+        plt.ylabel("Throughput(req/s)")
+        plt.xlabel("time(ms)")
+        plt.savefig("throughput_across_time.png")
+
+    def plot_latency_distribution(self, ms_separator: float, highest_vals=15):
+        """
+        Starting from minimum latency with ms_separator
+        step split the ms latency list in buckets
+        and plots the highest_vals top buckets
+        """
+        max_latency = max(self.ms_latency_list)
+        min_latency = min(self.ms_latency_list)
+
+        if max_latency < ms_separator:
+            LOG.remove()
+            LOG.add(
+                sys.stdout,
+                format="<red>[ERROR]:</red> {message}",
+            )
+
+            LOG.error(
+                f"Latency values are less than {ms_separator}, cannot produce latency distribution graph"
+            )
+            return
+
+        bins_number = (
+            int(
+                (max_latency - min_latency) // ms_separator
+                + bool((max_latency - min_latency) % ms_separator)
+            )
+            + 1
+        )
+
+        counts = [0] * bins_number
+        bins = [min_latency]
+        bin_val = min_latency
+        for _ in range(bins_number):
+            bin_val += ms_separator
+            bins.append(bin_val)
+
+        for lat in self.ms_latency_list:
+            counts[
+                int(
+                    (lat - min_latency) // ms_separator
+                    + bool((lat - min_latency) % ms_separator)
+                )
+            ] += 1
+
+        top_bins = []
+        top_counts = []
+        min_count = sorted(counts)[-highest_vals]
+        for ind in range(len(counts)):
+            if counts[ind] >= min_count:
+                top_bins.append(round(bins[ind], 3))
+                top_counts.append(counts[ind])
+
+        x_axis = range(len(top_bins))
+        plt.figure()
+        fig, ax = plt.subplots()
+        ax.bar(x_axis, top_counts, 0.9, align="center")
+        ax.set_xticks(x_axis)
+        ax.set_xticklabels(top_bins, rotation=25)
+        plt.ylabel("Count")
+        plt.xlabel("Latency")
+        fig.subplots_adjust(bottom=0.2)
+        plt.savefig("latency_distribution.png")
+
+
+def get_df_from_parquet_file(input_file: str):
+    return pd.read_parquet(input_file, engine="fastparquet")
+
+
+def default_analysis(send_file, response_file):
+    """
+    Produce the analysis results
+    """
+    analysis = Analyze()
+    df_sends = get_df_from_parquet_file(send_file)
+    df_responses = get_df_from_parquet_file(response_file)
+
+    successful_percent = analysis.iter_for_success_and_latency(df_sends, df_responses)
+
+    LOG.info(f"The request type sent is {analysis.get_req_type(df_responses)}")
+
+    print(
+        analysis.time_success_throughput_table(
+            df_sends, df_responses, successful_percent
+        )
+    )
+    print(analysis.latencies_table(df_sends, df_responses))
+
+    x = ["-"] * 20

```

Comment:
```suggestion
    x = "-" * 20
```
and then you can save on all the joins ;)